### PR TITLE
518 change material entity to return only 1 status per item if all materials were requested

### LINKED ISSUE #518: Change `material` entity to return only 1 status per item if all materials were requested
Currently, we load a huge amount of data during `getAllMaterials` request (all the materials and all the statuses). 

We should make it load the last `Status` per `getAllMeterials` request.

Could be done with `@AfterInsert` and a new field in the `material` table (e.g. `current_status`) to reduce database requests

### PR DESCRIPTION
**Pull request by IAlexHilimovich**
Closes #518

### CODE REVIEW

**Comment by ngame2011** [comment]
@IAlexHilimovich please resolve conflicts and request a review

**Review by Yoinezero — APPROVED** [review]

File: packages/apiserver/src/resources/persons/persons.loader.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    const personsMap: Record<number, Person> = persons.reduce((acc, curr) => {
+      acc[curr.id] = curr;
+      return acc;
+    }, {});
+
+    return ids.map((id) => personsMap[id]);
+  });
+};
+
+export default createPersonsLoader;
```
Comment (Chichmarin17) [inline packages/apiserver/src/resources/persons/persons.loader.ts:17]: Do we need `person.loader` at all? Didn't find any usages of that (except of registration within `app.module`)

File: packages/apiserver/src/resources/persons/person.schema.graphql [hunk 2]
```diff
@@ -13,6 +13,7 @@ type Person {
   location: [Location!]
   messages: [Message]
   states: [BlockedUsers]
+  materials: [Material]
```
Comment (Chichmarin17) [inline packages/apiserver/src/resources/persons/person.schema.graphql:16]: do we need this filed in our `gql` schema?

File: packages/frontapp/src/app/components/BookCardExtended/index.tsx [hunk 3]
```diff
@@ -147,9 +143,7 @@ const BookCardExtended: FC<IPropsBookCardExtended> = ({ item }) => {
             )}
           </TitleStatus>
           <TitleClaimHistory>Claim history:</TitleClaimHistory>
-          <HistoryBook>
-            was claimed {countOfHistoryClaimed || 0} times
-          </HistoryBook>
+          <HistoryBook>was claimed {item.claimCount || 0} times</HistoryBook>
```
Comment (Chichmarin17) [inline packages/frontapp/src/app/components/BookCardExtended/index.tsx:146]: could be great to add this hard-coded message to `localization` module

File: packages/apiserver/src/resources/persons/person.schema.graphql (hunk 2 above)
Comment (IAlexHilimovich) [reply packages/apiserver/src/resources/persons/person.schema.graphql:16]: Not needed, good catch

File: packages/apiserver/src/resources/persons/person.entity.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ import { Status } from '../statuses/status.entity';
 import { Message } from '../messages/message.entity';
 import { Location } from '../locations/location.entity';
 import { BlockedUsers } from '../blocked-users/blocked-users.entity';
+import { Material } from '../materials/material.entity';
```
[the remark this answers is not in the document]
Comment (IAlexHilimovich) [reply packages/apiserver/src/resources/persons/person.entity.ts]: Removed

File: packages/apiserver/src/resources/persons/persons.loader.ts (hunk 1 above)
Comment (IAlexHilimovich) [reply packages/apiserver/src/resources/persons/persons.loader.ts]: No longer used. Deleted it

File: packages/apiserver/src/resources/persons/person.schema.graphql (hunk 2 above)
Comment (IAlexHilimovich) [reply packages/apiserver/src/resources/persons/person.schema.graphql]: Deleted

**Review by Yoinezero — APPROVED** [review]

**Review by Chichmarin17 — APPROVED** [review]

**Pull request merged**